### ISSUE TITLE
Feature: ability to evaluate a string as a function

### ISSUE THREAD

**Issue by gesslar** [fluffos/fluffos#909]
It would be very handy to be able to store function expressions as a string which later we would be able to evaluate.

An example of use would be in templating, such as what I am attempting to do with help documentation. Ordinarily, if something requires functions to determine if parts or all of a help document is visible to the caller, we have to call the command object in question to determine this, however, if I could write in the help document template a string that is a function expression, this could be evaluated at call time to return what is needed. This would have lots of other use-cases as well. Such as storing this function string expression in properties that could be saved to file.

Such as: `"function(object who) { return who->query_gender() == \"male\" ; }`

**Comment by oiuv** [comment]
这个需求还是别加了，我感觉现在的功能可以满足

**Comment by jalbright015** [comment]
I agree with @oiuv. There are a number of existing ways you could achieve the same result.

**Comment by gesslar** [comment]
This is already achievable based on fallentree's updates. Just needs an
API.

**Comment by jalbright015** [comment]
Interesting.  Which update?

**Comment by gesslar** [comment]
I think it was this one. But I can't be positive, because I don't know how
to read it very well.

https://github.com/fluffos/fluffos/commit/bc042fc43077587460f4b8fd7ce4995e54079688

Brian M. Workman
***@***.***

**Comment by gesslar** [comment]
Hi @thefallentree , would you please expose an API to enable us to use the functionality you've added that permits evaluating a string? I tried so hard, but I could not get this to work.
